### PR TITLE
Add REST API specification for SAML APIs

### PR DESCRIPTION
**Pull request by jkakavas**
relates: elastic/elasticsearch-specification#3753

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_authenticate.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "security.saml_authenticate":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-saml-authenticate.html",
+      "description":"Exchanges a SAML Response message for an elasticsearch access token and refresh token pair"
+    },
+    "stability":"stable",
+    "visibility":"public",
+    "headers":{
+      "accept": [ "application/json"]
```
Comment (albertzaharovits) [inline rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_authenticate.json:10]:
```
     "content_type": ["application/json"]
```
I don't know if this is used.

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_sp_metadata.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    "url":{
+      "paths":[
+        {
+          "path":"/_security/saml/metadata/{realm_name}",
+          "methods":[
+            "GET"
+          ],
+          "parts":{
+            "name":{
```
Comment (albertzaharovits) [inline rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_sp_metadata.json:20]: Should be `realm_name` instead of `name`.

**Review by albertzaharovits — APPROVED** [review]

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_sp_metadata.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "security.saml_sp_metadata":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-saml-sp-metadata.html",
+      "description":"Generates SAML metadata for the elastic stack SAML 2.0 Service Provider"
```
Comment (sethmlarson) [inline rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_sp_metadata.json:5]: Should Elastic be capitalized here?

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_sp_metadata.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+  "security.saml_sp_metadata":{
```
Comment (sethmlarson) [inline rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_sp_metadata.json:2]: Thoughts on expanding sp to service_provider for better discoverability/readability?

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_authenticate.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+  "security.saml_authenticate":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-saml-authenticate.html",
+      "description":"Exchanges a SAML Response message for an elasticsearch access token and refresh token pair"
```
Comment (sethmlarson) [inline rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_authenticate.json:5]: nit: Elasticsearch should be capitalized?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Handful of nits and thoughts for you!

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_invalidate.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        {
+          "path":"/_security/saml/invalidate",
+          "methods":[
+            "POST"
+          ]
+        }
+      ]
+    },
+    "body":{
+      "description":"The LogoutRequest message",
```
[the remark this answers is not in the document]
Comment (jkakavas) [reply rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_invalidate.json:23]: No, why do you say so ?

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_authenticate.json (hunk 1 above)
Comment (jkakavas) [reply rest-api-spec/src/main/resources/rest-api-spec/api/security.saml_authenticate.json:10]: @sethmlarson should we remove this? We have it in a number of specs too, should we remove it from all in a follow up?

**Review by sethmlarson — APPROVED** [review]

**Pull request merged**